### PR TITLE
remove report v1 and dependencies

### PR DESCRIPTION
**Pull request by brendankenny**
part of #2276

also removes formatters, handlebars, handlebar build files, etc

still to do:
- decide what to do with PerfX. Its UI depends on injecting a handlebars partial into the report, which no longer supports handlebars partials. Disabled while we decide on that.
- move the v2 report to `lighthouse-core/report/`. Was too confusing to also do that in this PR :)

### CODE REVIEW

File: lighthouse-core/test/audits/deprecations-test.js [hunk 1]
```diff
@@ -69,8 +70,8 @@ describe('Console deprecations audit', () => {
     });
     assert.equal(auditResult.rawValue, false);
     assert.equal(auditResult.displayValue, '2 warnings found');
-    assert.equal(auditResult.extendedInfo.value.length, 2);
-    assert.equal(auditResult.extendedInfo.value[0].url, URL);
-    assert.equal(auditResult.extendedInfo.value[0].label, 'line: 123');
+    assert.equal(auditResult.details.items.length, 2);
+    assert.equal(auditResult.details.items[0][1].text, URL);
```
Comment (ebidel) [inline lighthouse-core/test/audits/deprecations-test.js]: ugh. we should probably rename `URL` while you're here. Node 8 has https://nodejs.org/api/url.html.

File: lighthouse-core/test/audits/dobetterweb/external-anchors-use-rel-noopener-test.js [hunk 2]
```diff
@@ -36,7 +36,7 @@ describe('External anchors use rel="noopener"', () => {
       URL: {finalUrl: URL},
```
Comment (ebidel) [inline lighthouse-core/test/audits/dobetterweb/external-anchors-use-rel-noopener-test.js]: more of them.

File: lighthouse-core/audits/consistently-interactive.js [hunk 3]
```diff
@@ -247,8 +246,7 @@ class ConsistentlyInteractiveMetric extends Audit {
           displayValue: Util.formatMilliseconds(timeInMs),
           optimalValue: this.meta.optimalValue,
           extendedInfo: {
-            value: extendedInfo,
-            formatter: Formatter.SUPPORTED_FORMATS.NULL,
+            value: extendedInfo
```
Comment (patrickhulce) [inline lighthouse-core/audits/consistently-interactive.js]: why you gotta nuke the extra commas? :(

Comment (brendankenny) [inline lighthouse-core/audits/consistently-interactive.js]: > why you gotta nuke the extra commas? :(

haha, I realized what I was doing halfway through and thought that might be a mistake. I'll restore them :)

File: lighthouse-core/test/audits/deprecations-test.js (hunk 1 above)
Comment (patrickhulce) [inline lighthouse-core/test/audits/deprecations-test.js]: you still have to `require` it though and it's meant to be identical so why need to rename?

Comment (ebidel) [inline lighthouse-core/test/audits/deprecations-test.js]: This is true. I suppose we'll know if someone introduces that require and things break. Just thinking ahead.

File: lighthouse-cli/printer.ts [hunk 4]
```diff
@@ -8,8 +8,8 @@
 /**
  * An enumeration of acceptable output modes:
  *   'json': JSON formatted results
- *   'html': An HTML report
- *   'domhtml': An HTML report rendered client-side with DOM elements
+ *   'html': An HTML report rendered client-side with DOM elements
+ *   'domhtml': Alias for 'html' report.
```
Comment (paulirish) [inline lighthouse-cli/printer.ts]: can you also tweak the filename for domhtml files to be just `.html`? That's in bin.ts

File: lighthouse-cli/printer.ts [hunk 5]
```diff
@@ -8,8 +8,8 @@
 /**
  * An enumeration of acceptable output modes:
  *   'json': JSON formatted results
- *   'html': An HTML report
- *   'domhtml': An HTML report rendered client-side with DOM elements
+ *   'html': An HTML report rendered client-side with DOM elements
```
Comment (brendankenny) [inline lighthouse-cli/printer.ts]: done

File: lighthouse-core/test/audits/deprecations-test.js (hunk 1 above)
Comment (brendankenny) [inline lighthouse-core/test/audits/deprecations-test.js]: we probably should generally use lowercase `url` for things like that

File: lighthouse-cli/printer.ts (hunk 4 above)
Comment (brendankenny) [inline lighthouse-cli/printer.ts]: done

**Comment by brendankenny** [comment]
PTAL

**Comment by ebidel** [comment]
Should the smoketests preserve `extendedInfo` for now?

**Comment by brendankenny** [comment]
> Should the smoketests preserve `extendedInfo` for now?

Yeah, I guess so, though there's no `details` expectations on master, which are arguably more important now

**Comment by brendankenny** [comment]
done

**Comment by brendankenny** [comment]
drive-by commenters happy?

**Comment by paulirish** [comment]
i'm happy.

**Comment by ebidel** [comment]
@brendankenny you handle file move?

**Comment by brendankenny** [comment]
> you handle file move?

got it now, thanks

**Comment by patrickhulce** [comment]
i'm good too

**Comment by brendankenny** [comment]
if you're developing Lighthouse, you'll also need to manually delete `lighthouse-core/report/partials/` and `lighthouse-core/report/templates/` as the built files in there were gitignored before and so aren't automatically deleted for you.

you'll probably also want to `yarn install-all` (and probably `yarn build-all`) to get rid of removed dependencies.

**Pull request merged**